_drafts/oryx/dependency-injection.md

```diff
@@ -0,0 +1,228 @@
+
+# Service Layer in Oryx
+
+The service layer in Oryx serves as the foundation for the business logic, with dependency injection (DI) playing a crucial role.
+
+The main objective of the service layer in Oryx is to abstract all of the system functionality, including querying backend services, data caching and reloading, state management, and reactivity.
+
+Furthermore, DI is utilized by many higher-level elements and concepts in Oryx, ranging from adapters and normalizers to HTTP interceptors, authentication, product loading, cart management, and the checkout process.
+
+## Npm package
+
+The `@spryker-oryx/di` package provides dependency injection not only to Oryx applications but also to other packages and can even be used outside the Oryx framework. It includes a variety of useful utilities to streamline working with dependency injection, such as:
+
+- TypeScript models definitions (e.g. Providers)
+- `Injector` - the implementation of a DI container in TypeScript
+- Functions used to set up and manage injectors (`createInjector()`, `getInjector()`, etc.)
+- Utilities that can be used to resolve dependencies (`inject` and `resolve`)
+
+In a typical Oryx application, the application orchestrator automatically handles much of the work related to setting up the DI container. Nevertheless, the `@spryker-oryx/di` library provides a set of tools that can be used to further customize and fine-tune the dependency injection as needed.
+
+# Using services
+
+## inject vs resolve
+
+There are two primary methods for injecting services and dependencies:
+
+- `inject` – usually used inside services
+- `resolve` – usually used inside (web) components
+
+### inject()
+
+```ts
+export class MyService {
+  constructor(protected otherService = inject(SomeOtherService)) {}
+}
+```
+
+The `inject` method is used to inject services from the current injector. This method can be used when the injection context is defined, such as in services at construction time or in provided factories that use `useFactory`.
+
+### resolve()
+
+```ts
+export class MyComponent extends LitElement {
+  protected myService = resolve(MyService);
+}
+```
+
+The `resolve` method works similarly to `inject`, but it tries to resolve the current injector using context (by default `globalThis`). This method is suitable to use in components, especially in web components, where the injection context may not be defined at construction time.
+
+### One container by default
+
+A typical Oryx application usually uses one global DI container, which is set up by the app orchestrator. This is currently the recommended approach for Oryx applications that utilize web components, as it provides a streamlined solution that typically addresses all requirements of the app.
+
+# Defining services
+
+In Oryx, we follow certain conventions for defining services.
+
+## String-Based Token
```

Comment:
Do we recommend the readers to use these conventions or is this just for reference?